src/main/java/io/kestra/plugin/surrealdb/SurrealDBConnection.java

```diff
@@ -0,0 +1,77 @@
+package io.kestra.plugin.surrealdb;
+
+import com.surrealdb.connection.SurrealConnection;
+import com.surrealdb.connection.SurrealWebSocketConnection;
+import com.surrealdb.driver.SyncSurrealDriver;
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.runners.RunContext;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Positive;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+public class SurrealDBConnection extends Task implements SurrealDBConnectionInterface {
+
+	@Builder.Default
+	private boolean useTls = false;
+
+	@NotNull
+	@Positive
+	private int port;
+
+	@NotNull
+	@NotBlank
+	private String host;
+
+	@NotNull
+	@NotBlank
+	private String username;
+
+	@NotNull
+	@NotBlank
+	private String password;
+
+	@NotNull
+	@NotBlank
+	private String namespace;
+
+	@NotNull
+	@NotBlank
+	private String database;
+
+	private SurrealConnection connection;
+
+	protected SyncSurrealDriver connect(RunContext context) throws IllegalVariableEvaluationException {
+		SurrealWebSocketConnection connection = new SurrealWebSocketConnection(host, port, useTls);
+		connection.connect(500);
```

Comment:
Is it connection timeout? Is it mandatory to set it?
If so it must be configurable via a task property with a sane default value.